qiskit_aer/backends/wrappers/aer_controller_binding.hpp

```diff
@@ -0,0 +1,461 @@
+
+#include "misc/warnings.hpp"
+DISABLE_WARNING_PUSH
+#include <pybind11/pybind11.h>
+DISABLE_WARNING_POP
+#if defined(_MSC_VER)
+    #undef snprintf
+#endif
+
+#include <vector>
+
+#include "framework/matrix.hpp"
+#include "framework/python_parser.hpp"
+#include "framework/pybind_casts.hpp"
+#include "framework/types.hpp"
+#include "framework/results/pybind_result.hpp"
+
+#include "controllers/aer_controller.hpp"
+
+#include "controllers/controller_execute.hpp"
+
+namespace py = pybind11;
+using namespace AER;
+
+template<typename T>
+class ControllerExecutor {
+public:
+    ControllerExecutor() = default;
+    py::object operator()(const py::handle &qobj) {
+#ifdef TEST_JSON // Convert input qobj to json to test standalone data reading
+        return AerToPy::to_python(controller_execute<T>(json_t(qobj)));
+#else
+        return AerToPy::to_python(controller_execute<T>(qobj));
+#endif
+    }
+
+    py::object execute(std::vector<Circuit> &circuits, Noise::NoiseModel &noise_model, AER::Config& config) const {
+        return AerToPy::to_python(controller_execute<T>(circuits, noise_model, config));
+    }
+};
+
+template<typename T>
+py::tuple write_value(size_t index, const optional<T> &v) {
+    return py::make_tuple(v.has_value(), v.value());
+}
+
+template<typename T>
+T write_value(size_t index, const T &v) {
+    return v;
+}
+
+template<typename T>
+void read_value(const py::tuple &t, size_t index, optional<T> &v) {
+    if (t[index].cast<py::tuple>()[0].cast<bool>())
+        v.value(t[index].cast<py::tuple>()[1].cast<T>());
+}
+
+template<typename T>
+void read_value(const py::tuple &t, size_t index, T &v) {
+    v = t[index].cast<T>();
+}
+
+template<typename MODULE>
+void bind_aer_controller(MODULE m) {
+    py::class_<ControllerExecutor<Controller> > aer_ctrl (m, "aer_controller_execute");
+    aer_ctrl.def(py::init<>());
+    aer_ctrl.def("__call__", &ControllerExecutor<Controller>::operator());
+    aer_ctrl.def("__reduce__", [aer_ctrl](const ControllerExecutor<Controller> &self) {
+        return py::make_tuple(aer_ctrl, py::tuple());
+    });
+    aer_ctrl.def("execute", [aer_ctrl](ControllerExecutor<Controller> &self,
+                                       std::vector<Circuit> &circuits,
+                                       py::object noise_model,
+                                       AER::Config &config) {
+
+        Noise::NoiseModel noise_model_native;
+        if (noise_model)
+          noise_model_native.load_from_json(noise_model);
+
+        return self.execute(circuits, noise_model_native, config);
+    });
+
+    py::class_<Config> aer_config(m, "AerConfig");
+    aer_config.def(py::init());
+    aer_config.def_readwrite("shots", &Config::shots);
+    aer_config.def_readwrite("method", &Config::method);
+    aer_config.def_readwrite("device", &Config::device);
+    aer_config.def_readwrite("precision", &Config::precision);
+    // executor=None,
+    // max_job_size=None,
+    // max_shot_size=None,
```

Comment:
Are these needed? Or is it just to document the python side fields we're not specifying here, if so can you put a comment above that says "missing options from python" or something.